x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/action/TransportDeleteLifecycleAction.java

```diff
@@ -74,7 +75,8 @@ public ClusterState execute(ClusterState currentState) {
                         }
                         ClusterState.Builder newState = ClusterState.builder(currentState);
                         IndexLifecycleMetadata currentMetadata = currentState.metaData().custom(IndexLifecycleMetadata.TYPE);
-                        if (currentMetadata.getPolicyMetadatas().containsKey(request.getPolicyName()) == false) {
+                        if (currentMetadata == null
```

Comment:
since this is an anonymous class, we do not have good unit test coverage, so I've added a yaml test to walk this branch